Improve `ui_combo_12` ui for custom color presets

crumblingstatue opened the pull request:
Shows color swatches for both the preset color and the resulting token

![image](https://github.com/user-attachments/assets/028320eb-8623-4830-8e59-1b7f01ce8264)

crumblingstatue committed:
Improve `ui_combo_12` ui for custom color presets

Shows color swatches for both the preset color and the resulting token

crumblingstatue commented on src/lib.rs:
```diff
@@ -191,22 +191,45 @@ impl Colorix {
+                            self.update_color(ctx, i);
                         }
-                    });
+                        let tokens = [
```
This is kinda questionable, but I'm not sure how else to get the resulting tokens

frankvgompel replied:
Quite. I'll think about it.

crumblingstatue replied:
> Quite. I'll think about it.

If you figure some API out, I can rebase my pull request on top of it

frankvgompel replied:
You can remove this and instead of `token[i]` you should be able to use the new `self.theme[i].get_token(i)`

crumblingstatue replied:
Done

crumblingstatue commented on src/tokens.rs:
```diff
@@ -214,4 +214,31 @@ impl ColorPreset {
             Self::Custom([r, g, b]) => [r, g, b],
         }
     }
+    pub(crate) const fn label(self) -> &'static str {
```
I added this so I can make `Custom` label in the dropdown plain, without messing with the `Debug` impl.

frankvgompel replied:
Can we replace DROPDOWN_TEXT with `self.themes[i].label()`?

crumblingstatue replied:
> Can we replace DROPDOWN_TEXT with `self.themes[i].label()`?

Done

frankvgompel replied:
On 2nd thought: leave it as it was.

crumblingstatue replied:
> On 2nd thought: leave it as it was.

Can you explain? Do you want me to restore the `DROPDOWN_TEXT` array (that I removed in a commit)?
What's the rationale behind that?

crumblingstatue committed:
Improve the presentation for non-custom presets

It makes them align with custom-presets, and adds the token
color preview for them.

crumblingstatue committed:
Add option to show rgb labels for custom color presets

crumblingstatue committed:
Remove DROPDOWN_TEXT in favor of ColorPreset::label

crumblingstatue committed:
Only set egui visuals when toggling light/dark mode

crumblingstatue committed:
Use self.tokens.get_token(i) rather than duplicated tokens array

crumblingstatue committed:
Remove rgb labels in favor of copying theme to clipboard

frankvgompel commented:
Abolutely brilliant work. Thanks.
